src/Invictus.Testing/LogicAppsProvider.cs

```diff
@@ -0,0 +1,307 @@
+        /// <param name="logicAppName">The name of the logic app resource running in Azure.</param>
+        /// <param name="authentication">The authentication mechanism to authenticate with Azure.</param>
+        /// <param name="logger">The instance to write diagnostic trace messages while interacting with the provider.</param>
+        public static LogicAppsProvider LocatedAt(
+            string resourceGroup,
+            string logicAppName,
+            LogicAuthentication authentication,
+            ILogger logger)
+        {
+            Guard.NotNullOrWhitespace(resourceGroup, nameof(resourceGroup));
+            Guard.NotNullOrWhitespace(logicAppName, nameof(logicAppName));
+            Guard.NotNull(authentication, nameof(authentication));
+            
+            logger = logger ?? NullLogger.Instance;
+            return new LogicAppsProvider(resourceGroup, logicAppName, authentication, logger);
+        }
+
+        /// <summary>
+        /// Sets the start time when the logic app runs were executed.
+        /// </summary>
+        /// <param name="startTime">The date that the logic app ran.</param>
+        public LogicAppsProvider WithStartTime(DateTime startTime)
+        {
+            // TODO: just migrated from original 'helper' class, should be with offset?
+            _startTime = startTime;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the time period in which the retrieval of the logic app runs should succeed.
+        /// </summary>
+        /// <param name="timeout">The period to retrieve logic app runs.</param>
+        public LogicAppsProvider WithTimeout(TimeSpan timeout)
+        {
+            _timeout = timeout;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the correlation ID as client tracking when retrieving logic app runs.
+        /// </summary>
+        /// <param name="correlationId">The client tracking of the logic app runs.</param>
+        public LogicAppsProvider WithCorrelationId(string correlationId)
+        {
+            Guard.NotNull(correlationId, nameof(correlationId));
+
+            _hasCorrelationId = true;
+            _correlationId = correlationId;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the tracking property as filter when retrieving logic app runs.
+        /// </summary>
+        /// <param name="trackedPropertyName">The name of the tracked property to filter on.</param>
+        /// <param name="trackedPropertyValue">The value of the tracked property to filter on.</param>
+        public LogicAppsProvider WithTrackedProperty(string trackedPropertyName, string trackedPropertyValue)
+        {
+            Guard.NotNull(trackedPropertyName, nameof(trackedPropertyName));
+            Guard.NotNull(trackedPropertyValue, nameof(trackedPropertyValue));
+
+            _hasTrackedProperty = true;
+            _trackedPropertyName = trackedPropertyName;
```

Comment:
Maybe we should switch to a dictionary so that folks can look for multiple properties that are matching?

I'm sure this would be interesting, or not @mbraekman ?

Reply:
Totally!
Would be perfect to track down a specific logic app within an entire flow.

I am wondering, do you adjust the current method, or would you create an additional method to support the dictionary? This way devs don't have to create a dictionary for a single keyValue-pair?

Reply:
I would personally keep the current API and expect folks to call it multiple times since it's a fluent model. Internally we should just move to a dictionary instead.

Reply:
Can be good for a next issue so we can list the options.

Reply:
#33